packages/web/src/components/PopupModal/PopupModal.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+} from '@chakra-ui/react';
+
+interface PopupModalProps {
+  openModalText: String;
+  header: String;
+  body: String;
+  openButtonVariant?: 'link' | 'outline' | (string & {}) | 'ghost' | 'solid' | 'unstyled';
+  onConfirm?: () => Promise<void>;
+  succussMessage?: string;
+  errorMessage?: string;
+}
+
+export const PopUpModal: React.FC<PopupModalProps> = ({
+  openModalText,
+  header,
+  openButtonVariant,
+  body,
+  onConfirm,
+  succussMessage,
+  errorMessage,
+}) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [modalHeader, setModalHeader] = useState(header);
+  const [bodyText, setBodyText] = useState(body);
+  const [isLoading, setIsLoading] = useState(false);
+  const [confirmButtonVisible, setConfirmButtonVisible] = useState(true);
+
+  return (
+    <>
+      <Button
+        onClick={() => {
+          setModalHeader(header);
+          setBodyText(body);
+          setConfirmButtonVisible(true);
+          onOpen();
+        }}
+        variant={openButtonVariant}
+      >
+        {openModalText}
+      </Button>
+
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>{modalHeader}</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>{bodyText}</ModalBody>
+
+          <ModalFooter>
+            <Button colorScheme="blue" mr={3} onClick={onClose}>
+              Close
+            </Button>
+            {onConfirm && confirmButtonVisible && (
+              <Button
+                isLoading={isLoading}
+                onClick={async () => {
+                  try {
+                    setIsLoading(true);
+                    await onConfirm();
+                    if (succussMessage) {
+                      setModalHeader('Success !!!');
+                      setBodyText(succussMessage);
+                    }
+                  } catch (e) {
+                    if (errorMessage) {
+                      setModalHeader('An Error Occurred');
+                      setBodyText(errorMessage);
+                    }
+                  } finally {
+                    setIsLoading(false);
+                    setConfirmButtonVisible(false);
+                  }
+                }}
+              >
+                Confirm
+              </Button>
+            )}
+          </ModalFooter>
```

Comment:
A couple changes here
- Wrapped the buttons in an HStack so they have standard spacing
- The close button is the secondary action, so it should be more subdued than the submit button
- Instead of `boolean && React.Element` we should use a ternary operator, `boolean ? React.Element : null`, so React can discard the result faster and not check if `false` is a valid thing that should be shown. Also easier to see when the checks end
- Moved the logic of the `onClick` to a separate function so the UI is easier to look through

```suggestion
          <ModalFooter>
            <HStack justifyContent="flex-end">
              <Button variant="ghost" onClick={onClose}>
                Close
              </Button>
              {onConfirm && confirmButtonVisible ? (
                <Button
                  isLoading={isLoading}
                  colorScheme="blue"
                  onClick={onConfirmClick}
                >
                  Confirm
                </Button>
              ) : null}
            </HStack>
          </ModalFooter>
```